// payload/dist/auth/operations/me.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const url_1 = __importDefault(require("url"));
const jsonwebtoken_1 = __importDefault(require("jsonwebtoken"));
const getExtractJWT_1 = __importDefault(require("../getExtractJWT"));
async function me({ req, collection, }) {
    const extractJWT = (0, getExtractJWT_1.default)(req.payload.config);
    let response = {
        user: null,
    };
    if (req.user) {
        const parsedURL = url_1.default.parse(req.originalUrl);
        const isGraphQL = parsedURL.pathname === `/api${req.payload.config.routes.graphQL}`;
        const user = await req.payload.findByID({
            id: req.user.id,
            collection: collection.config.slug,
            req,
            depth: isGraphQL ? 0 : collection.config.auth.depth,
            overrideAccess: false,
            showHiddenFields: false,
        });
        if (req.user.collection !== collection.config.slug) {
            return {
                user: null,
            };
        }
        response = {
            user,
            collection: req.user.collection,
        };
        const token = extractJWT(req);
        if (token) {
            response.token = token;
            const decoded = jsonwebtoken_1.default.decode(token);
            if (decoded)
                response.exp = decoded.exp;
        }
    }
    // /////////////////////////////////////
    // After Me - Collection
    // /////////////////////////////////////
    await collection.config.hooks.afterMe.reduce(async (priorHook, hook) => {
        await priorHook;
        response = await hook({
            req,
            response,
            context: req.context,
        }) || response;
    }, Promise.resolve());
    return response;
}
exports.default = me;
